aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -156,22 +167,49 @@ Tensor empty_like(const Tensor& self) {
   return native::empty_like(self, self.options());
 }
 
-Tensor empty_like(const Tensor& self, const TensorOptions& options) {
+Tensor empty_like(
+    const Tensor& self,
+    const TensorOptions& options,
+    c10::optional<c10::MemoryFormat> optional_memory_format) {
+
+  TORCH_CHECK(
+      !(options.layout() == kSparse &&
+          optional_memory_format.has_value()),
+      " memory format options is incompatible with sparse tensors");
```

Comment:
nit: arguably, only `kStrided` has layouts, everything else doesn't have layouts. In particular, if someone gave you an MKL DNN layout tensor, you'd also want to bug out if you got a layout.

Reply:
Changed message and check.